libs/react/src/lib/modal/modal.tsx

```diff
@@ -18,26 +18,26 @@ export interface ModalProps {
   onDismiss?: ModalEventListener
 }
 
-const ModalHeader = ({ header = '', onClose }: Partial<ModalProps>) => {
+const ModalHeader = memo(({ header = '', onClose }: Partial<ModalProps>) => {
   const handleClose: ModalEventListener = (event) => {
     if (onClose) onClose(event)
   }
   return (
     <div className="header">
-      <h3>{header}</h3>
+      <h3 id="modal_header">{header}</h3>
```

Comment:
Yes, you should generate a uuid and store in a useState, so that every instance of the component gets a unique ID. The you can do ```id={`modal_header_${uuid}`}``` or similar.

Here's an example of this in another component: https://github.com/sebgroup/green/blob/main/libs/react/src/lib/accordion/accordion.tsx

Reply:
done! thanks for the example! helped a lot :)